syne_tune/optimizer/schedulers/multiobjective/linear_scalarizer.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from collections.abc import Iterable
+from itertools import groupby
+from typing import Dict, Any, List, Union, Callable, Optional
+
+import numpy as np
+
+from syne_tune.backend.trial_status import Trial
+from syne_tune.optimizer.scheduler import TrialScheduler, TrialSuggestion
+from syne_tune.optimizer.schedulers.fifo import FIFOScheduler
+
+logger = logging.getLogger(__name__)
+
+
+def _all_equal(iterable: Iterable) -> bool:
```

Comment:
Looks pretty cryptic. Why not something like
`a = lst[0]; all(x == a for x in lst[1:])`

Reply:
The implementation below returns True for lists with length 0 and works for any iterable, also those that do not support indexing. It might be overkill for this application but its probably good ot have more versatile tooling in the library.